### DISCUSSION TITLE
Intersection/Crossing points between two point clouds or between pyvista.line and spline?

### DISCUSSION THREAD

**Discussion by abbas-tari** [pyvista/pyvista#1825] [Q&A]
Hi - I am wondering how I can find crossing points of two point clouds or  between high resolution line and spline in pyvista?

I can convert one point cloud to a surface, so in this case the question would be crossing points between the surface and a high resolution line?

Thanks

**Comment by akaszynski** [comment]
Can you post a minimum working example here?

**Reply by abbas-tari** [reply]
Sure. Here is a very simple example: the idea is to find intersection points between line and the polygon defined by four points. `collision`, `boolean_difference`, and `intersection` methods won't work for some reasons such as triangulation etc. I also tried `ray_trace` with no success. However, I am aware of `scipy` or `pykdtree` as you have explained it here:
https://github.com/pyvista/pyvista-support/issues/107, which seems a good idea for a very high resolution point clouds not a polygon with a few points like below.

```` 
import pyvista as pv
import numpy as np

# simplified point cloud (four points of polygon)
turningPoints = np.array([[17.11, 0, 242.805], [-3.99, 0, 266.185] ,  [-9.45, 0,262.755],  [-10.095, 0, 247.56]] )

# Cell
xSecCell = pv.PolyData(turningPoints, [4, 0, 1, 2, 3])
xSecMesh = xSecCell.triangulate()

# Spline
xSecSpline = pv.Spline(turningPoints)


pointa = [-20, 0, 260]
pointb = [20, 0, 260]
lineDivider = pv.Line(pointa, pointb, resolution= 1000)

plotter = pv.Plotter()
plotter.add_mesh(xSecMesh)
plotter.add_mesh(lineDivider)
plotter.show()
````
